src/main/java/org/jenkinsci/test/acceptance/docker/DockerRule.java

```diff
@@ -53,11 +55,27 @@ public DockerRule(Class<T> type) {
         return this;
     }
 
+    /**
+     * The rule cause the test to be skipped when docker is not available on the host. Setting this to true will cause
+     * the tests to fail.
```

Comment:
I do not think this is a good idea. The test should not know about the environment it runs in. Rather, a CI job (or whatever) should be configured to highlight skipped tests.

Reply:
I understand that users to this tooling might feel that way but for testing this repo, I thing this is essential. When I started digging in this, both containers was unbuildable without anyone noticing...

---

I do not insist doing it this way. Though, what is wrong with test author declaring the docker based tests are so essential for the project that when they can not run they should fail the build?

Reply:
> both containers was unbuildable without anyone noticing

Trying to address in #15.

Reply:
> what is wrong with test author declaring the docker based tests are so essential for the project that when they can not run they should fail the build?

Just feels like something that should be defined either in the CI job, or in the build script (e.g., Surefire configuration), or using something like JUnit categories. The same requirement could apply to _any_ kind of test which might be skipped under some conditions (missing Internet connection, ad nauseam).

Reply:
Would a java property be any better? I am happy with the PR as it is now ...